engine/src/main/java/org/hibernate/validator/cfg/AnnotationDef.java

```diff
@@ -110,14 +109,14 @@ protected C addAnnotationAsParameter(String key, AnnotationDef<?, ?> value) {
 		return getThis();
 	}
 
-	protected A createAnnotationProxy() {
-		AnnotationDescriptor<A> annotationDescriptor = new AnnotationDescriptor<>( annotationType );
+	protected AnnotationDescriptor<A> createAnnotationDescriptor() {
+		AnnotationDescriptor.Builder<A> annotationDescriptorBuilder = new AnnotationDescriptor.Builder<>( annotationType );
```

Comment:
With that nice builder in place, did you think about using the builder as field directly and get rid of some/all of the intermediary maps?

Reply:
Not sure I understand what you're implying here?

Reply:
What I mean is, can we e.g. get rid of the `parameters` map and work with a builder directly, instead of first adding everything to the maps and then creating + feeding a builder from the end? For sure not strictly needed, but it may further simplify this code here. I'll leave it to you :)

Reply:
I don't think it buys us much but done.